32-explore_pathway_separation.Rmd

```diff
@@ -0,0 +1,416 @@
+---
+title: "Pathway 'separation'"
+output:   
+  html_notebook: 
+    toc: true
+    toc_float: true
+---
+
+**J. Taroni 2018**
+
+We've noted that pathways or genesets that are similar to one another, such as 
+cell types of the myeloid lineage (e.g., neutrophils and monocytes/macrophages),
+can tend to get "lumped together" in PLIER models trained on training sets with
+smaller sample sizes or less relevant biological contexts.
+In contrast, the MultiPLIER model (~37K samples) tends to "separate" similar 
+pathways into different latent variables.
+(See `06-sle-wb_cell_type` and `07-sle_cell_type_recount2_model`.)
+
+**Here, we're interested in the notion of "pathway separation" when different 
+datasets are used for training.**
+
+We'll group related pathways/genesets into pathway sets.
+For instance, we'll group monocyte- and macrophage-related gene sets such as
+`SVM Monocytes` and `DMAP_MONO2` into a "monocyte-related pathway set."
+
+We define **pathway separation** as the following:
+
+* For each pathway set, at least one latent variable is significantly
+associated (`FDR < 0.05`) with a pathway in that set.
+We consider capturing _only one pathway set_ to be insufficient for this 
+evaluation.
+In essence, we want to make sure each set of pathways is represented in or 
+captured by the model. 
+* Each pathway set is _uniquely_ and significantly associated with at least one
+latent variable. 
+In the neutrophil vs. monocyte/macrophage example, it's not sufficient to have a
+neutrophil-associated latent variable and a latent variable associated with 
+both neutrophils and monocytes/macrophages; we also want the monocyte/macrophage
+set to be captured in a latent variable that is _not also associated_ with
+the neutrophil set.
+
+## Set up
+
+```{r}
+library(ComplexHeatmap)
+```
+
+### Functions
+
+```{r}
+`%>%` <- dplyr::`%>%`
+```
+
+#### Custom function for detecting pathway separation
+
+```{r}
+CheckPathwaySeparation <- function(plier.model, pathway.set1, pathway.set2,
```

Comment:
I imagine the separation aspect is highly confounded by overlapping genes in input genesets too

Reply:
This is a good point. I've thought about this a little bit.
* Perhaps it is then a higher bar when separation occurs if there's a non-negligible amount of overlap between "similar" gene sets. If we take the IFN pathways as an example, there is significant overlap (see these [MSigDB results](http://software.broadinstitute.org/gsea/msigdb/compute_overlaps.jsp?geneSetName=REACTOME_INTERFERON_ALPHA_BETA_SIGNALING&collection=CP:REACTOME)) from genes that encode things like transcription factors (e.g., _IRF7_). The differences are the signaling molecules themselves and receptors, etc. which I assuming is what the model is picking up on.
* One approach would be to pre-process the _input gene sets_ via something like crosstalk correction ([Donato, et al.](http://doi.org/10.1101/gr.153551.112), [Chen, et al.](https://doi.org/10.1186/s13040-018-0175-7)), but maybe having the large body of expression data + the model "sort it out" is more appealing.
* "Do the loadings (`Z`) 'get rid of' the overlap?" could be interesting to look at at some point in the future.

I think this experiment does address the question I've set out to address, which is "Does the size of the training set or the biological conditions/contexts that comprise the training set effect a PLIER model's ability to distinguish similar pathways?"

But definitely somethings to think about or improve down the line and/or when other folks in the lab are looking into disentanglement of features.